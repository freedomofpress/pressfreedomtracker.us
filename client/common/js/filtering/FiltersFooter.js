import React from 'react'
import queryString from 'query-string'
import { HorizontalLoader } from '~/filtering/Loader'


function FiltersFooter({
	handleClearFilters,
	loading,
	pageFetchParams,
	filtersTouched,
	exportPath,
}) {
	const params = pageFetchParams
	const qs = queryString.stringify(params)
	const exportUrl = `${exportPath}?${qs}`
	return (
		<div className="filters__footer">
			<div className="filters__text filters__text--dim filters__text--meta">
				Need to do more complex filtering?
				{' '}
				<a href={exportUrl} className="filters__link">Download the Data.</a>
			</div>

			<span className="filters__button-toolbar">
				<button
					className="filters__button"
					onClick={handleClearFilters}
					type="button"
				>
					Clear Filters
				</button>

				<button
					className="filters__button filters__button--bordered filters__button--wide"
					disabled={!filtersTouched}
					type="submit"
				>
					{loading > 0 && <HorizontalLoader />}
					{loading === 0 && 'Apply Filters'}
				</button>
			</span>
		</div>
	)
}


export default FiltersFooter
