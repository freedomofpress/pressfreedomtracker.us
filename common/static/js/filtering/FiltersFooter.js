import React from 'react'
import { HorizontalLoader } from '~/filtering/Loader'


function FiltersFooter({
	handleApplyFilters,
	handleClearFilters,
	loading,
	filtersTouched,
}) {
	return (
		<div className="filters__footer">
			<div className="filters__text filters__text--dim filters__text--meta">
				Need to do more complex filtering?
				{' '}
				<a href="#" className="filters__link">Download the Data.</a>
			</div>

			<span className="filters__button-toolbar">
				<button
					className="filters__button"
					onClick={handleClearFilters}
				>
					Clear Filters
				</button>

				<button
					className="filters__button filters__button--bordered filters__button--wide"
					onClick={handleApplyFilters}
					disabled={!filtersTouched}
				>
					{loading > 0 && <HorizontalLoader />}
					{loading === 0 && 'Apply Filters'}
				</button>
			</span>
		</div>
	)
}


export default FiltersFooter
