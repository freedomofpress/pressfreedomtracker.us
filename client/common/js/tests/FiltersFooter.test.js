import React from 'react';
import { shallow, mount } from 'enzyme';
import FiltersFooter from '~/filtering/FiltersFooter';
import {
	CATEGORIES_FILTER,
	PAGE_FETCH_PARAMS
} from '~/tests/factories/CategoriesFilterFactory';

describe('FiltersFooter', () => {
	let categories;
	let pageFetchParams;

	beforeEach(() => {
		categories = CATEGORIES_FILTER;
		pageFetchParams = PAGE_FETCH_PARAMS;
	});

	afterEach(() => {
		categories = null;
		pageFetchParams = null;
	});

	it('should render without error in "debug" mode', () => {
		const component = shallow(
		<FiltersFooter
			categories={categories}
		/>
		);

		expect(component.debug()).toMatchSnapshot();
	});

	it('should render the correct export link without param', () => {
		const component = mount(
			<FiltersFooter
				categories={categories}
				exportPath={'test-path/'}
			/>
		);

		// Check the download link is rendered properly
		expect(component.find('a.filters__link').length).toEqual(1);
		expect(component.find('a.filters__link').text()).toEqual('Download the Data.');
		expect(component.find('a.filters__link').filterWhere((item) => {
			return item.prop('href') === 'test-path/export/?'
		}).length).toEqual(1);
	});

	it('should render the correct export link with param', () => {
		const component = mount(
			<FiltersFooter
				categories={categories}
				exportPath={'test-path/'}
				pageFetchParams={pageFetchParams}
			/>
		);

		// Check the download link is rendered properly
		expect(component.find('a.filters__link').length).toEqual(1);
		expect(component.find('a.filters__link').text()).toEqual('Download the Data.');
		expect(component.find('a.filters__link').filterWhere((item) => {
			return item.prop('href') === 'test-path/export/?categories=5&charged_under_espionage_act=True'
		}).length).toEqual(1);
	});

	it('should render filter footer buttons properly ', () => {
		const component = mount(
			<FiltersFooter
				categories={categories}
				loading={0}
			/>
		);

		expect(component.find('.horizontal-loader').length).toEqual(0);

		//Check 2 filter buttons exist
		expect(component.find('button.filters__button').length).toEqual(2);

		// Check the clear button exists and it's type is button
		expect(component.find('button.filters__button').filterWhere((item) => {
			return item.prop('type') === 'button'
		}).length).toEqual(1);
		expect(component.find('button.filters__button').filterWhere((item) => {
			return item.prop('type') === 'button'
		}).text()).toEqual('Clear Filters');

		// Check the apply button exists and it's type is submit and it is bordered
		expect(component.find('button.filters__button--bordered').filterWhere((item) => {
			return item.prop('type') === 'submit'
		}).length).toEqual(1);
		expect(component.find('button.filters__button--bordered').filterWhere((item) => {
			return item.prop('type') === 'submit'
		}).text()).toEqual('Apply Filters');
	});

	it('should render filter footer buttons properly ', () => {
		const component = mount(
			<FiltersFooter
				categories={categories}
				loading={1}
			/>
		);
		expect(component.find('.horizontal-loader').length).toEqual(1);
	});
});
