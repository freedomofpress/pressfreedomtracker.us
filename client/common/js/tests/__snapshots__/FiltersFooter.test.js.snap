// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersFooter should render without error in "debug" mode 1`] = `
"<div className=\\"filters__footer\\">
  <div className=\\"filters__text filters__text--dim filters__text--meta\\">
    Need to do more complex filtering?
     
    <a href=\\"undefinedexport/?\\" className=\\"filters__link\\">
      Download the Data.
    </a>
  </div>
  <span className=\\"filters__button-toolbar\\">
    <button className=\\"filters__button\\" onClick={[undefined]} type=\\"button\\">
      Clear Filters
    </button>
    <button className=\\"filters__button filters__button--bordered filters__button--wide\\" disabled={true} type=\\"submit\\" />
  </span>
</div>"
`;
